enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationLimitService.java

```diff
@@ -0,0 +1,101 @@
+import com.wultra.app.onboardingserver.errorhandling.IdentityVerificationException;
+import com.wultra.app.onboardingserver.errorhandling.OnboardingProcessLimitException;
+import com.wultra.security.powerauth.client.model.error.PowerAuthClientException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static com.wultra.app.onboardingserver.impl.service.ActivationFlagService.ACTIVATION_FLAG_VERIFICATION_IN_PROGRESS;
+
+/**
+ * Service for checking identity verification limits.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+@Service
+public class IdentityVerificationLimitService {
+
+    private static final Logger logger = LoggerFactory.getLogger(IdentityVerificationLimitService.class);
+
+    private final IdentityVerificationRepository identityVerificationRepository;
+    private final IdentityVerificationConfig identityVerificationConfig;
+    private final OnboardingProcessRepository onboardingProcessRepository;
+    private final ActivationFlagService activationFlagService;
+
+    /**
+     * Service constructor.
+     * @param identityVerificationRepository Identity verification repository.
+     * @param identityVerificationConfig Identity verification config.
+     * @param onboardingProcessRepository Onboarding process service.
+     * @param activationFlagService Activation flag service.
+     */
+    @Autowired
+    public IdentityVerificationLimitService(IdentityVerificationRepository identityVerificationRepository, IdentityVerificationConfig identityVerificationConfig, OnboardingProcessRepository onboardingProcessRepository, ActivationFlagService activationFlagService) {
+        this.identityVerificationRepository = identityVerificationRepository;
+        this.identityVerificationConfig = identityVerificationConfig;
+        this.onboardingProcessRepository = onboardingProcessRepository;
+        this.activationFlagService = activationFlagService;
+    }
+
+    /**
+     * Check attempt limit of failed identity verifications. Fail onboarding process in the attempt count has exceeded the limit.
+     * @param ownerId Owner identification.
+     * @throws PowerAuthClientException Thrown when activation flag request fails.
+     * @throws IdentityVerificationException Thrown in case identity verification is invalid.
+     * @throws OnboardingProcessLimitException Thrown when maximum failed attempts for identity verification have been reached.
+     */
+    public void checkIdentityVerificationLimit(OwnerId ownerId) throws PowerAuthClientException, IdentityVerificationException, OnboardingProcessLimitException {
+        List<IdentityVerificationEntity> identityVerifications = identityVerificationRepository.findByActivationIdOrderByTimestampCreatedDesc(ownerId.getActivationId());
```

Comment:
Maybe not clear that `findByActivationIdOrderByTimestampCreatedDesc` returns all so far failed identity verifications. The sorting is not needed here at all.

But obviously it will work
- when checking the verification limit on creation, reset

Reply:
It returns both `FAILED` and `IN_PROGRESS` verifications, we are interested in the total number. The sorting is used just for consistency with other methods, in case we need to get this list anywhere else, but I can remove it.

Reply:
Aaah, then `maxFailedAttempts` naming is a little bit confusing, maybe `maxAttempts`. But for sure a small comment above the line would be enough too.

Reply:
I will review this and add some comments to make it clear, thanks.